vault.go

```diff
@@ -108,32 +111,33 @@ func addVault(pod *corev1.Pod, namespace, serviceAccountToken string, databases
 			vaultContainer.Args = append(vaultContainer.Args, "--job")
 		}
 
-		patch = append(patch, patchOperation{
-			Op:    "add",
-			Path:  "/spec/containers/-",
-			Value: vaultContainer,
-		})
+		pod.Spec.Containers = append(pod.Spec.Containers, vaultContainer)
 
 		initContainer.Args = append(initContainer.Args, "--init")
 		initContainer.Name = initContainer.Name + "-init"
-		var init interface{}
-
-		initPath := "/spec/initContainers"
-		if len(pod.Spec.InitContainers) != 0 || inited == true {
-			initPath = initPath + "/-"
-			init = initContainer
-		} else {
-			init = []corev1.Container{initContainer}
-			inited = true
-		}
+		initContainers = append(initContainers, initContainer)
+	}
 
-		patch = append(patch, patchOperation{
-			Op:    "add",
-			Path:  initPath,
-			Value: init,
-		})
+	var initOp string
+	if len(pod.Spec.InitContainers) != 0 {
+		initContainers = append(initContainers, pod.Spec.InitContainers...)
+		initOp = "replace"
+	} else {
+		initOp = "add"
 	}
 
+	patch = append(patch, patchOperation{
+		Op:    "replace",
+		Path:  "/spec/containers",
+		Value: pod.Spec.Containers,
+	})
+
+	patch = append(patch, patchOperation{
```

Comment:
You can combine these two `append` calls if you want to as it its type is `append([]T, T...)`

Reply:
yeah could do